hugegraph-api/src/main/java/com/baidu/hugegraph/api/graph/VertexAPI.java

```diff
@@ -114,6 +115,70 @@ public String create(@Context GraphManager manager,
         });
     }
 
+    /**
+     * TODO: Adapter for param("createIfNotExist") or delete it?
+     * batch update steps like:
+     * 1. get all newVertices' ID & combine first
+     * 2. get all oldVertices & update
+     * 3. add the final vertex together
+     **/
+    @PUT
+    @Timed
+    @Decompress
+    @Path("batch")
+    @Consumes(APPLICATION_JSON)
+    @Produces(APPLICATION_JSON_WITH_CHARSET)
+    public String update(@Context HugeConfig config,
+                         @Context GraphManager manager,
+                         @PathParam("graph") String graph,
+                         VertexRequest req) {
+        LOG.debug("Graph [{}] update vertices: {}", graph, req.jsonVertices);
+        checkUpdatingBody(req.jsonVertices);
+
+        HugeGraph g = graph(manager, graph);
+        checkBatchSize(config, req.jsonVertices);
+        // Not support automatic Id strategy now (check once?)
+        E.checkArgument(g.vertexLabel(req.jsonVertices.get(0).label).
+                        idStrategy() != IdStrategy.AUTOMATIC,
+                        "Automatic Id strategy is not supported now");
+
+        Map<Id, JsonVertex> maps = new HashMap<>(req.jsonVertices.size());
+
+        return this.commit(config, g, maps.size(), () -> {
+            /*
+            * 1.Put all newVertices' properties into map (combine first)
+            * - Consider primary-key & user-define ID mode first
+            * */
+            req.jsonVertices.forEach(newVertex -> {
+                String labelId = g.vertexLabel(newVertex.label).id().asString();
+                Id id = this.getVertexId(g, labelId, newVertex);
+                this.updateExistElement(newVertex, maps.get(id),
```

Comment:
Is the parameter reversed?  I think it should be this.updateExistElement(maps.get(id), newVertex, req.updateStrategies)